tephra-core/src/main/java/co/cask/tephra/persist/TransactionSnapshot.java

```diff
@@ -44,49 +45,56 @@
   public TransactionSnapshot(long timestamp, long readPointer, long writePointer, Collection<Long> invalid,
                              NavigableMap<Long, TransactionManager.InProgressTx> inProgress,
                              Map<Long, Set<ChangeId>> committing, Map<Long, Set<ChangeId>> committed) {
+    this(timestamp, readPointer, writePointer, invalid, inProgress);
+    this.committingChangeSets = committing;
+    this.committedChangeSets = committed;
+  }
+
+  public TransactionSnapshot(long timestamp, long readPointer, long writePointer, Collection<Long> invalid,
+                             NavigableMap<Long, TransactionManager.InProgressTx> inProgress) {
     this.timestamp = timestamp;
     this.readPointer = readPointer;
     this.writePointer = writePointer;
     this.invalid = invalid;
     this.inProgress = inProgress;
-    this.committingChangeSets = committing;
-    this.committedChangeSets = committed;
+    this.committingChangeSets = Collections.emptyMap();
+    this.committedChangeSets = Collections.emptyMap();
   }
 
-  /**
-   * Returns the timestamp from when this snapshot was created.
-   */
   public long getTimestamp() {
     return timestamp;
   }
 
-  /**
-   * Returns the read pointer at the time of the snapshot.
-   */
   public long getReadPointer() {
     return readPointer;
   }
 
-  /**
-   * Returns the next write pointer at the time of the snapshot.
-   */
   public long getWritePointer() {
     return writePointer;
   }
 
-  /**
-   * Returns the list of invalid write pointers at the time of the snapshot.
-   */
   public Collection<Long> getInvalid() {
     return invalid;
   }
 
+  public NavigableMap<Long, TransactionManager.InProgressTx> getInProgress() {
```

Comment:
Can you add `@Override` annotation to all above methods implementing `TransactionVisibilityState` interface?